cli/pcluster/config/pcluster_config.py

```diff
@@ -282,11 +282,11 @@ def __init_region(self):
         """
         Evaluate region to use and set in the environment to be available for all the boto3 calls.
 
-        Order is 1) AWS_DEFAULT_REGION env 2) Config file 3) default from mapping
+        Order is 1) AWS_DEFAULT_REGION env 2) Config file 3) default from aws config file
         """
         if os.environ.get("AWS_DEFAULT_REGION"):
             self.region = os.environ.get("AWS_DEFAULT_REGION")
-        else:
+        elif self.region:
             os.environ["AWS_DEFAULT_REGION"] = self.region
```

Comment:
here we need another `else` block, to set the default value in the environment `os.environ["AWS_DEFAULT_REGION"] = boto3sessionregion`

The region in the environment should be always set, to have a consistent behaviour and always have a value when calling the `get_region` utility function.